base64/base64.go

```diff
@@ -1,12 +1,113 @@
 // Package base64 implements base64 encoding as specified by RFC 4648.
 package base64
 
+import (
+	"strings"
+)
+
+const base64 = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/"
+
 // Encode encodes src using base64 encoding.
+// SGVsbG8sIHdvcmxk
 func Encode(src []byte) (dst []byte) {
-	return nil
+	switch len(src) {
+	case 3:
+		return []byte{
+			base64[src[0]>>2],
+			base64[(src[0]&0b00000011)<<4|src[1]>>4],
+			base64[(src[1]&0b00001111)<<2|src[2]>>6],
+			base64[src[2]&0b00111111],
+		}
+
+	case 2:
+		return []byte{
+			base64[src[0]>>2],
+			base64[(src[0]&0b00000011)<<4|src[1]>>4],
+			base64[(src[1]&0b00001111)<<2],
+			'=',
+		}
+	case 1:
+		return []byte{
+			base64[src[0]>>2],
+			base64[(src[0]&0b00000011)<<4],
+			'=',
+			'=',
+		}
+	default:
+		if len(src) != 0 {
+			return append([]byte{
+				base64[src[0]>>2],
+				base64[(src[0]&0b00000011)<<4|src[1]>>4],
+				base64[(src[1]&0b00001111)<<2|src[2]>>6],
+				base64[src[2]&0b00111111],
+			}, Encode(src[3:])...)
```

Comment:
Please do not use recursion here.

1. This was not a topic in our lessons.
2. This is not efficient. For large inputs, your program will run out of stack size.
3. There is an obvious solution with a for-loop here.
4. We asked you not use the recursion already.